src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -153,7 +153,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Public Overrides Function InstrumentReturnStatement(original As BoundReturnStatement, rewritten As BoundStatement) As BoundStatement
             Dim previous As BoundStatement = MyBase.InstrumentReturnStatement(original, rewritten)
             If Not original.IsEndOfMethodReturn Then
-                Return AddDynamicAnalysis(original, previous)
+                If original.ExpressionOpt IsNot Nothing Then
+                    ' Synthesized return statements require instrumentation if they return values,
+                    ' e.g. in simple expression lambdas.
```

Comment:
Yes, this case is specifically for expression lambdas--previously their bodies were not being instrumented. IsEndOfMethodReturn marks the return statement synthesized at the bottom of a VB function, and these are not instrumented. I do not know of a case were the expression value would be Nothing, but it is technically optional.